app/knight.py

```diff
@@ -0,0 +1,33 @@
+from typing import Dict, Any
+from app.armour import Armour
+from app.weapon import Weapon
+from app.potion import Potion
+
+
+class Knight:
+    def __init__(self, data: Dict[str, Any]) -> None:
+        self.name = data["name"]
+        self.hp = data["hp"]
+        self.power = data["power"]
+        self.armour = Armour(
+            data["armour"]
+        ) if "armour" in data else Armour([])
+        self.weapon = Weapon(
+            data["weapon"]["name"],
+            data["weapon"]["power"]
+        )
```

Comment:
You are directly accessing the `weapon` key from the `data` dictionary which may raise a KeyError if the key is not present. It's better to use the `get` method which will return `None` if the key is not present. This applies to other similar cases in your code as well.